sqlserver/datadog_checks/sqlserver/agent_history.py

```diff
@@ -0,0 +1,180 @@
+        j.name AS job_name,
+        sjh1.job_id,
+        sjh1.step_name,
+        sjh1.step_id,
+        sjh1.instance_id AS step_instance_id,
+        (
+            SELECT MIN(sjh2.instance_id)
+            FROM msdb.dbo.sysjobhistory AS sjh2
+            WHERE sjh2.job_id = sjh1.job_id
+            AND sjh2.step_id = 0
+            AND sjh2.instance_id >= sjh1.instance_id
+        ) AS completion_instance_id,
+        (
+            SELECT DATEDIFF(SECOND, GETDATE(), SYSDATETIMEOFFSET()) +
+                DATEDIFF(SECOND, '19700101',
+                    DATEADD(HOUR, sjh1.run_time / 10000,
+                        DATEADD(MINUTE, (sjh1.run_time / 100) % 100,
+                            DATEADD(SECOND, sjh1.run_time % 100,
+                                CAST(CAST(sjh1.run_date AS CHAR(8)) AS DATETIME)
+                            )
+                        )
+                    )
+                )
+        ) AS run_epoch_time,
+        (
+            (sjh1.run_duration / 10000) * 3600
+            + ((sjh1.run_duration % 10000) / 100) * 60
+            + (sjh1.run_duration % 100)
+        ) AS run_duration_seconds,
+        CASE sjh1.run_status
+            WHEN 0 THEN 'Failed'
+            WHEN 1 THEN 'Succeeded'
+            WHEN 2 THEN 'Retry'
+            WHEN 3 THEN 'Canceled'
+            WHEN 4 THEN 'In Progress'
+            ELSE 'Unknown'
+        END AS step_run_status,
+        sjh1.message
+    FROM
+        msdb.dbo.sysjobhistory AS sjh1
+    INNER JOIN msdb.dbo.sysjobs AS j
+    ON j.job_id = sjh1.job_id
+    ORDER BY completion_instance_id DESC
+)
+SELECT * FROM HISTORY_ENTRIES
+WHERE
+    EXISTS (
+        SELECT 1
+        FROM msdb.dbo.sysjobhistory AS sjh2
+        WHERE sjh2.instance_id = HISTORY_ENTRIES.completion_instance_id
+        AND (DATEDIFF(SECOND, GETDATE(), SYSDATETIMEOFFSET()) +
+                    DATEDIFF(SECOND, '19700101',
+                        DATEADD(HOUR, sjh2.run_time / 10000,
+                            DATEADD(MINUTE, (sjh2.run_time / 100) % 100,
+                                DATEADD(SECOND, sjh2.run_time % 100,
+                                    CAST(CAST(sjh2.run_date AS CHAR(8)) AS DATETIME)
+                                )
+                            )
+                        )
+                    )
+                + (sjh2.run_duration / 10000) * 3600
+        + ((sjh2.run_duration % 10000) / 100) * 60
+        + (sjh2.run_duration % 100)) > 0 + {last_collection_time_filter}
+    )
+"""
+
+
+def agent_check_getter(self):
+    return self._check
+
+
+class SqlserverAgentHistory(DBMAsyncJob):
+    def __init__(self, check, config: SQLServerConfig):
+        # do not emit any dd.internal metrics for DBM specific check code
+        self.tags = [t for t in check.tags if not t.startswith('dd.internal')]
+        self.log = check.log
+        self._config = config
+        collection_interval = float(self._config.agent_jobs_config.get('collection_interval', 15))
```

Comment:
`15` is different from the default `60` seconds in config yaml. should we change it to 60s?

Reply:
Keeping it at 15 and changing in the config yaml since the duration is within constraints

Reply:
that works as long as they are consistent